### ISSUE TITLE
redundant wallet not picking up transactions

### ISSUE THREAD

**Issue by Erovyr101** [decred/dcrwallet#28]
I have a wallet from which I did a few ticket purchases. when I do dcrctl ... --wallet gettickets 1,  I can see the list of all my immature tickets.

In order to avoid the risk of missing a vote I  started 2 other instance of dcrwallet on other systems. using the same seed. On creation, dcrd and the wallets will sync up to the most recent block and I can see the same tickets and transactions.

However if I make new ticket purchase from my first system (only that one is used for purchase).  The 2 other wallets will randomly or not see the new ticket even 6 hours later.

When I check the dcrd and dcrwallet window on all system they all have processed the most recent block (624 at this time). If I take the transaction id from my main wallet I can verify that it was successfully received with the mainnet web block explorer. Restarting dcrwallet on the 2 redundant systems will sometime add one or 2 new transaction but most of the time not.

If I use the transaction id and do a  search on all systems,  the one on which the purchase was made will see the transaction, and the others will find no record of that transaction even if they have processed the corresponding block recently or a while back.

The only way I have found to force a wallet to sync up is to delete it and recreate it. On creation it will sync everything and then it will report all the tickets and transactions.

Also, I am not sure this issue is only affecting redundant wallets because after 350 blocks were created yesterday, the wallet I created at launch with my previous seed still couldn't see my airdrop even though the transactions was listed in the block explorer and all daemons were telling me they were up to date with the latest block.  It's only after I deleted my wallet and recreated it that I was able to see the transaction and correct balance.

All systems use v0.0.4, are on windows system and on different networks

**Comment by chappjc** [comment]
This would seem to be by design.  See davecgh's response [here](https://forum.decred.org/threads/block-explorer-and-web-wallet.446/page-3#post-6796) and [here](https://forum.decred.org/threads/block-explorer-and-web-wallet.446/page-3#post-6809).

**Comment by Erovyr101** [comment]
I see, so that's why I the only transaction trace I can find is when using getrawtransaction. Because the wallet is just ignoring it otherwise. In that case what would be the correct way to re-sync other wallets (create new addresses according to the main wallet growth ?  set the wallet to reuse addresses when purchasing tickets ?Maybe a "full rescan" command would be useful if someone would loose count of how many adresses have been created by the main wallet.

Then, what would be the correct way to have redundant wallets online so they can vote  in case the main wallet is down or a network error occurs when a ticket is chosen ?. From my perspective the only way to achieve that was to create identical wallet online but having just one used for transactions and purchases.  The other wallet would just be unlocked and ready to vote but with a really high maintainbalance value and low ticket price so it doesn't create new tickets.

**Comment by chappjc** [comment]
The devs have given fairly stern advice not to use multiple wallets generated with the same seed.

However, I believe with the `--ticketaddress` flag for dcrwallet, you can avoid having multiple wallets with the same seed and achieve staking redundancy.  Sorry for the [wiki link](https://wiki.decred.org/Solo_Stake_Mining#Splitting_the_Hot.2FCold_Wallet).

I'd love if a developer could provide some advice on how to go about setting up redundancy with PoS.

**Comment by Erovyr101** [comment]
Yes I think further advice would be needed because the --ticketaddress flag seems to allow a separation of vote from the main wallet such as : Wallet A is a cold wallet (locked or offline) but has defined with --ticketaddress that wallet B  can vote in its place.  Which is great but doesn't provide redundancy by itself.

**Comment by cjepson** [comment]
You can use redundant wallets so long as the only function of the wallet is voting. You must also either reuse or pregenerate all the addresses on the voting wallet, so it knows what addresses to look out for when you send it tickets.

Ideally, after creating your voting wallet and generating addresses, close the wallet with 'stop' and then copy it.

**Comment by chappjc** [comment]
Suppose then you reopen this copied wallet on multiple machines (each with enablestakemining still on?), then when it comes time to vote won't there be a race?

**Comment by cjepson** [comment]
No. Stake voting is handled differently and doesn't rely on address tracking. As long as --enablestakemining is on and the wallet is unlocked, it will vote automatically and without issue. You may get an error that the vote was already seen on the network, but no corruption.

**Comment by Erovyr101** [comment]
Thanks for the information, that clears up a lot of things. However on a more practical note, I'm not sure copying wallet files between systems is the most secure or efficient way of handling redundancy. 

For the moment I've been experimenting with following procedure on each redundant system:
1- init a wallet using your seed  (--create)
2- before starting the dcrwallet, copy wallet.db to keep an initial empty version of itself ex: wallet.db.init
3- start dcrwallet

Then When you want a full rescan on one of the redundant wallet to keep track of possible new addresses:
1- stop dcrwallet
2- copy wallet.db.init over wallet.db
3- start dcrwallet

The wallet will print "No recorded transaction history -- needs full rescan",  and the wallet will sync and become up to date with all transactions and addresses from your main wallet. This way you don't have to move wallet files manually or re-enter the seed each time.

I guess if you could initiate the same "full rescan" operation with a dcrctl wallet command that would be an even better way of dealing with this situation.

**Comment by cjepson** [comment]
Yeah, creating the wallet and copying pre-full scan is an easy way to ensure it does the initial rescan upon loading it up. There should be a way to trigger that from RPC as well, but for now it can only be triggered on startup. 

Be sure to know that anything that is imported, like a redeem script for a multsig P2SH, will NOT be automatically discovered by a wallet rescan. Be careful not to lose imported keys or scripts.
